File: plugin/zeppelin/src/main/scala/org/intellij/plugin/zeppelin/idea/toolwindow/actions/RefreshInterpretersAction.scala
package jetbrains.zeppelin.idea.toolwindow.actions

import com.intellij.icons.AllIcons
import com.intellij.openapi.actionSystem.{AnActionEvent, Presentation}
import com.intellij.openapi.project.DumbAwareAction
import org.intellij.plugin.zeppelin.components.ZeppelinComponent

/**
  * Refresh a list of available interpreters on Zeppelin.
  */
class RefreshInterpretersAction extends DumbAwareAction {
  val templatePresentation: Presentation = getTemplatePresentation
  templatePresentation.setIcon(AllIcons.Actions.Refresh)
  templatePresentation.setText("Refresh Interpreters list")

  override def actionPerformed(event: AnActionEvent): Unit = {
    val connection = ZeppelinComponent.connectionFor(event.getProject)
    connection.updateInterpreterList()
  }
}

File: zeppelin/src/org/intellij/plugin/zeppelin/idea/toolwindow/actions/RefreshInterpretersAction.kt
package org.intellij.plugin.zeppelin.idea.toolwindow.actions

import com.intellij.icons.AllIcons
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.project.DumbAwareAction
import org.intellij.plugin.zeppelin.components.ZeppelinComponent
import org.intellij.plugin.zeppelin.constants.ZeppelinConstants
import org.intellij.plugin.zeppelin.utils.ZeppelinLogger

/**
 * Refresh a list of available interpreters on Zeppelin.
 */
class RefreshInterpretersAction :
        DumbAwareAction("Refresh", "Refresh Zeppelin connection", AllIcons.Actions.Refresh) {

    override fun actionPerformed(event: AnActionEvent) {
        val project = event.project
        if (project == null) {
            ZeppelinLogger.printError(ZeppelinConstants.ERROR_PROJECT_IS_NOT_SPECIFIED)
            return
        }
        ZeppelinComponent.connectionFor(project).updateInterpreterList(true)
    }
}